libs/MeshKernel/src/ConnectMeshes.cpp

```diff
@@ -63,6 +65,9 @@ void meshkernel::ConnectMeshes::AreEdgesAdjacent(const Mesh2D& mesh,
         {
             endNode = mesh.m_edges[edge2].second;
         }
+
+        double absCosPhi = std::abs(NormalizedInnerProductTwoSegments(edge1Start, edge1End, edge2Start, edge2End, mesh.m_projection));
+        areAdjacent = (absCosPhi > 0.9 && absCosPhi <= 1.0);
```

Comment:
Can the bounds be also (optional) function parameters?

Reply:
I can do this but I think that the algorithm is quite fragile and, as can be shown with the separation fraction, any small change results in garbage being generated.

But I will certainly make a named constant